code/game/machinery/bank_machine.dm

```diff
@@ -95,14 +89,61 @@
 	switch(action)
 		if("siphon")
 			say("Siphon of station credits has begun!")
-			siphoning = TRUE
+			start_siphon()
 			. = TRUE
 		if("halt")
 			say("Station credit withdrawal halted.")
-			end_syphon()
+			end_siphon()
 			. = TRUE
 
-/obj/machinery/computer/bank_machine/proc/end_syphon()
+/obj/machinery/computer/bank_machine/proc/start_siphon()
+	siphoning = TRUE
+
+/obj/machinery/computer/bank_machine/proc/end_siphon()
+	if(!siphoning || !siphoning_credits)
+		return
 	siphoning = FALSE
-	new /obj/item/holochip(drop_location(), syphoning_credits) //get the loot
-	syphoning_credits = 0
+	new /obj/item/holochip(drop_location(), siphoning_credits) //get the loot
+	siphoning_credits = 0
+
+
+/obj/machinery/computer/bank_machine/process(delta_time)
+	..()
+	if(!siphoning)
+		return
```

Comment:
```suggestion
		return PROCESS_KILL
```
or even better, replace `siphoning` with process handling
start processing in start_siphon() and stop siphoning in end_siphon() and when you need to check if its siphoning `datum_flags & DF_ISPROCESSING`